app/models/spree/order.rb

```diff
@@ -30,7 +30,7 @@ class Order < ApplicationRecord
       go_to_state :complete
     end
 
-    attr_accessor :use_billing, :checkout_processing
+    attr_accessor :use_billing, :checkout_processing, :save_bill_address, :save_ship_address
```

Comment:
I wonder if this should be handled by the controller but this solution is very short and simple.

Reply:
Yeah I went back and forth on the question of where and how this should be handled. It's at a point where we're updating the order through nested attributes and it involves the order, the customer, the user, and potentially multiple addresses. I initially tried having the controller handle it but it had to juggle a lot of different objects. It wasn't nearly as elegant.

The other consideration here is that I wanted to put it in an `after_commit` callback so it's handled after the main transaction has completed (which is when the addresses are actually persisted).

What do you think?

What do you think?

Reply:
I still like your solution. Spree overdid the callback usage but it's okay in moderation.